### ISSUE TITLE
Add a custom plugin encoder for all DateTime parameters

### ISSUE THREAD

**Issue by remi-stripe** [stripe/stripe-dotnet#1164]
Timestamps in the API are represented as DateTime in stripe-dotnet. Those have to be converted to a timestamp internally such as [here](https://github.com/stripe/stripe-dotnet/pull/1163) which is easy to forget.

It would be better to add a custom plugin encoder for those parameters as we do for other types [here](https://github.com/stripe/stripe-dotnet/tree/master/src/Stripe.net/Infrastructure/Middleware/ParserPlugin).

**Comment by KClough** [comment]
Agreed, I'm seeing this issue with the TosAcceptanceDate field. Is there currently any workaround?

**Comment by remi-stripe** [comment]
@KClough I just tried to pass `TosAcceptanceDate` on the latest version of stripe-dotnet and it works fine for me:

```
            var accountOptions = new StripeAccountCreateOptions() {
                Email = "bob@example.com",
                Type = StripeAccountType.Custom,
                Country = "FR",
                TosAcceptanceDate = DateTime.UtcNow.Date,
                TosAcceptanceIp = "8.8.8.8",
                TosAcceptanceUserAgent = "user-agent-7",
            };

            var accountService = new StripeAccountService();
            StripeAccount account = accountService.Create(accountOptions);            var accountOptions = new StripeAccountCreateOptions() {
                Email = "bob@example.com",
                Type = StripeAccountType.Custom,
                Country = "FR",
                TosAcceptanceDate = DateTime.UtcNow.Date,
                TosAcceptanceIp = "8.8.8.8",
                TosAcceptanceUserAgent = "user-agent-7",
            };

            var accountService = new StripeAccountService();
            StripeAccount account = accountService.Create(accountOptions);
```

In the dashboard log for that request creation I then see:
```
{
  country: "FR",
  type: "custom",
  email: "bob@example.com",
  tos_acceptance: {
    date: "1524700800",
    ip: "8.8.8.8",
    user_agent: "user-agent-7"
  }
}
```

Do you have more details about the issue you are encountering?

**Comment by KClough** [comment]
@remi-stripe Thanks, error was on my end, I'm good now.

**Comment by ob-stripe** [comment]
This was implemented in #1283 and will be released with #1293.

**Comment by ob-stripe** [comment]
Implemented in 20.0.0.
